File: B3WM.Flutter/lib/models/pivot_storage_item.dart
/// Snapshot estático do Pivot Tradicional (issue #14), espelhando o
/// `ExtremeStorageItem`: o backend calcula de forma pura (sem estado ao
/// vivo) e o app só exibe linhas horizontais do dia atual/último pregão.
class PivotLevel {
  final String key;
  final double value;

  PivotLevel({required this.key, required this.value});

  factory PivotLevel.fromJson(Map<String, dynamic> json) => PivotLevel(
        key: json['key'] as String? ?? '',
        value: (json['value'] as num?)?.toDouble() ?? 0,
      );
}

class PivotStorageItem {
  final String symbol;
  final DateTime? date;
  final String source;
  final double high;
  final double low;
  final double close;
  final int lineCount;
  final List<PivotLevel> levels;

  PivotStorageItem({
    this.symbol = '',
    this.date,
    this.source = '',
    this.high = 0,
    this.low = 0,
    this.close = 0,
    this.lineCount = 2,
    this.levels = const [],
  });

  double levelValue(String key) {
    for (final l in levels) {
      if (l.key == key) return l.value;
    }
    return double.nan;
  }

  factory PivotStorageItem.fromJson(Map<String, dynamic> json) =>
      PivotStorageItem(
        symbol: json['symbol'] as String? ?? '',
        date: json['date'] != null
            ? DateTime.tryParse(json['date'] as String)
            : null,
        source: json['source'] as String? ?? '',
        high: (json['high'] as num?)?.toDouble() ?? 0,
        low: (json['low'] as num?)?.toDouble() ?? 0,
        close: (json['close'] as num?)?.toDouble() ?? 0,
        lineCount: (json['lineCount'] as num?)?.toInt() ?? 2,
        levels: (json['levels'] as List<dynamic>? ?? [])
            .map((e) => PivotLevel.fromJson(e as Map<String, dynamic>))
            .toList(),
      );
}

File: B3WM.Shared/Models/PivotStorageItem.cs
namespace B3WM.Shared.Models
{
    /// <summary>
    /// Nível de Pivot Tradicional (Profit): P, R1..R5, S1..S5.
    /// Serializado em camelCase para o Flutter (levels[].key/value).
    /// </summary>
    public class PivotLevel
    {
        public string Key { get; set; } = "";
        public double Value { get; set; }
    }

    /// <summary>
    /// Snapshot estático do Pivot do dia (só dia atual/último pregão).
    /// Intraday: fonte HLC de D-1. Daily: fonte HLC da semana anterior.
    /// Função pura — sem estado ao vivo, sem SignalR.
    /// </summary>
    public class PivotStorageItem
    {
        public string Symbol { get; set; } = "";
        /// <summary>Sessão exibida (dia atual ou último pregão com dados).</summary>
        public DateTime Date { get; set; }
        /// <summary>"D-1" (intraday) ou "W-1" (daily).</summary>
        public string Source { get; set; } = "";
        public double High { get; set; }
        public double Low { get; set; }
        public double Close { get; set; }
        public int LineCount { get; set; } = 2;
        public List<PivotLevel> Levels { get; set; } = new();
    }
}
